docs/migration.rst

```diff
@@ -0,0 +1,89 @@
+Zarr-Python 3 Migration Guide
+=============================
+
+Zarr-Python 3 introduces a number of changes to the API, including a number
+of significant breaking changes and pending deprecations.
+
+This page provides a guide highlighting the most notable changes to help you
+migrate your code from Zarr-Python 2.x to Zarr-Python 3.x.
+
+Compatibility target
+--------------------
+
+Zarr-Python 3 represents a major refactor of the Zarr-Python codebase. Some of the goals motivating this refactor included:
+
+  - adding support for the V3 specification (alongside the V3 specification)
+  - cleaning up internal and user facing APIs
+  - improving performance (particularly in high latency storage environments like cloud object store)
+
+Though these goals necessitated some breaking changes to the API (hence the major version update), we have tried to maintain
+backwards compatibility in the most widely used parts of the API including the `Array` and `Group` classes and the top-level
+API (e.g. `zarr.open_array` and `zarr.open_group`). It is worth noting that we significantly evolved the internal data model,
+moving away from a model that was tightly coupled to the v2 specification, and to a more generic representation of Zarr objects.
```

Comment:
This sentence raises more questions (why did we evolve the data model? how did we evolve the data model? what does the new model look like?), and I don't think is needed for a simple compatability guide, so I think best to drop?
```suggestion
API (e.g. `zarr.open_array` and `zarr.open_group`).
```